vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -819,7 +836,7 @@ impl<T: DeviceBacking> DriverWorkerTask<T> {
         cpu: u32,
     ) -> anyhow::Result<IoIssuer> {
```

Comment:
I'll leave it up to the storage folks to decide if they want to take this, but I think our preferred approach would be to convert these functions to returning full thiserror enums for all the different cases and then just matching on that, instead of using downcast_ref.